src/history/hs.c

```diff
@@ -750,9 +750,17 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_RECONCILE *r, WT_M
                         WT_ERR(__hs_insert_record(session, cursor, btree_id, key, upd,
                           WT_UPDATE_MODIFY, modify_value, stop_ts_pair));
                         __wt_scr_free(session, &modify_value);
-                    } else
+                    } else {
                         WT_ERR(__hs_insert_record(session, cursor, btree_id, key, upd,
                           WT_UPDATE_STANDARD, full_value, stop_ts_pair));
+                        /*
+                         * If we are evicting, we can now free some older updates which has already
+                         * been written to the history store. However, we can only free the updates
+                         * after a full update inserted into the history store as a full update.
```

Comment:
Can shorten to something like "If we are evicting, we can now free older updates which have been written to the history store. However, we can only free after a full update has been inserted into the history store."

Reply:
This is not exactly what I want to say. We can free only the full inserted into history store is originally a full update. We may insert a full update and it's originally a modify.

Reply:
I'll update the comment to clarify that.